src/platform/plugins/shared/data_views/common/data_views/data_views.ts

```diff
@@ -1320,6 +1320,8 @@ export class DataViewsService {
       .then((response) => {
         indexPattern.id = response.id;
         indexPattern.version = response.version;
+        this.dataViewCache.set(indexPattern.id, Promise.resolve(indexPattern as DataView));
+        this.dataViewLazyCache.set(indexPattern.id, Promise.resolve(indexPattern as DataViewLazy));
```

Comment:
We don't need to load the list or clean the cache in Discover, if the cache is also updated when a data view is updated, e.g. change of name.